app/controllers/notifications_controller.rb

```diff
@@ -0,0 +1,21 @@
+class NotificationsController < ApplicationController
+  before_action :authenticate_user!
+
+  def index
+    @notifications = current_user.message_notifications.unread
+    @has_read_notifications = current_user.message_notifications.read.any?
+  end
+
+  def update
+    notifications = current_user.message_notifications.unread
+    @notification = Notification.find(params[:id])
+
+    if notifications.include?(@notification)
+      @notification.mark_as_read!
+    end
```

Comment:
This is handling permissions/authorization, right? Like, if I `PATCH` to this route for your notification it shouldn't mark it as read?

If that's the case, then I'd like to replace this logic with a pundit authorization policy. `app/controllers/messages_controller.rb:7` and `app/policies/messaging_policy.rb:6` are a good example of how we can move that logic into a new class like the rest of the codebase.

Reply:
The notification update (mark as read) logic is moving to conversations in my next major commit on this PR so I don’t think this work is necessary. This whole method is disappearing.

Reply:
Gotcha, sounds good.